clang-tidy-checks/setup.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+function info() {
+  echo "[info]: " $1
+}
+
+function success() {
+  echo "success!"
+}
+
+function check_requirements() {
+  info "checking requirements"
+  cmake --version &&
+  gcc --version &&
+  python3 --version &&
+  ninja --version &&
+  lld --version &&
+  success
+}
+
+function clone_llvm() {
+  info "cloing llvm"
+  git clone https://github.com/llvm/llvm-project.git &&
+  git fetch --all --tags &&
+  git checkout -b tags/llvmorg-11.0.0 &&
+  success
+}
+
+function apply_patches() {
+  info "applying patches"
+  cd llvm-project
+  patch potential-unbounded-loop-check.diff &&
```

Comment:
another question: I have this repo cloned in `~/github/pytorch/test-infra`, and I'm trying running this command from within my llvm/llvm-project clone (on commit `176249bd6732`):
```sh
patch ~/github/pytorch/test-infra/clang-tidy-checks/potential-unbounded-loop-check.diff
```
however, it's taking a long time and not give any output; if you write a `clang-tidy-checks/README.md`, could you indicate how long this command is expected to take?

Reply:
ok, I tried using `git apply` instead of `patch` and that seems to have worked; could you make that same replacement in `clang-tidy-checks/setup.sh`? also, it gave these warnings, in case that's relevant:
```
/Users/sestep/github/pytorch/test-infra/clang-tidy-checks/potential-unbounded-loop-check.diff:9: new blank line at EOF.
+
/Users/sestep/github/pytorch/test-infra/clang-tidy-checks/potential-unbounded-loop-check.diff:3051: new blank line at EOF.
+
warning: 2 lines add whitespace errors.
```

Reply:
Resolved below - I didn't pass in the directory in which the patch should be applied. This causes it to hang because it's waiting for input from stdin